### PR TITLE
Update nav_list to support FontAwesome icons on the left bar

### PR DESCRIPTION
**Pull request by Kralizek**
This is an enhancement or feature.

## Summary

Added the possibility to support icons in the navigation list on the left bar.

```
about-me:
  - title: "About me"
    icon: "fas fa-info-circle"
    url: /about/
  - title: "Resume"
    icon: "far fa-fw fa-file-pdf"
    url: "assets/files/resume.pdf"
```

## Context

When adding a navigation list on the left bar, it's now possible to add an icon using FontAwesome. [Here is a live example](https://renatogolia.com/about/)

### CODE REVIEW

**Review by iBug — APPROVED** [review]
Looks good.

**Comment by Kralizek** [comment]
What's next?

**Comment by iBug** [comment]
Sorry I don't know what next, maybe just wait patiently. Only @mmistakes can merge PRs (I don't have access). This one is good to be merged IMO.

File: _includes/nav_list [hunk 1]
```diff
@@ -7,10 +7,13 @@
   <ul class="nav__items">
     {% for nav in navigation %}
       <li>
+        {% if nav.icon %}
+            {% capture icon %}<i class="{{ nav.icon }}" aria-hidden="true"></i> {% endcapture %}
+        {% endif %}
         {% if nav.url %}
-          <a href="{{ nav.url | relative_url }}"><span class="nav__sub-title">{{ nav.title }}</span></a>
```
Comment (lsolesen) [inline _includes/nav_list:11]: Shouldn't this keep the relative_url filter?

Comment (iBug) [reply _includes/nav_list:11]: This `nav_url` comes from nowhere - am I mistaken?

Comment (Kralizek) [reply _includes/nav_list:11]: Honestly I don't remember it and it puzzles me a lot.

**Pull request closed without merging**